src/main/java/uk/gov/register/serialization/RSFExecutor.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.register.serialization;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import uk.gov.register.core.HashingAlgorithm;
+import uk.gov.register.core.Item;
+import uk.gov.register.core.Register;
+import uk.gov.register.util.HashValue;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+public class RSFExecutor {
+
+    private Map<String, RegisterCommandHandler> registeredHandlers;
+
+
+    public RSFExecutor() {
+        registeredHandlers = new HashMap<>();
+    }
+
+    public void register(RegisterCommandHandler registerCommandHandler) {
+        registeredHandlers.put(registerCommandHandler.getCommandName(), registerCommandHandler);
+    }
+
+    public RSFResult execute(RegisterSerialisationFormat rsf, Register register) {
+        // this instead of integer can be full command
+        // then when it's referenced just empty the value
+        // when checking for orphan there should not be any hash with command as those
+        // should have been emptied
+        // ?? memory consumption here ??
+
+        // for now hash and line from file?
+        // hash and rsf file line number
+        Map<String, Integer> shaRefLine = new HashMap<>();
```

Comment:
didn't do that for the memory consumption concern as it's a bigger class, but can change that